lib/src/Uncertainty/Algorithm/MetaModel/Kriging/openturns/GaussianProcessConditionalCovariance.hxx

```diff
@@ -0,0 +1,129 @@
+
+BEGIN_NAMESPACE_OPENTURNS
+
+/**
+ * @class GaussianProcessConditionalCovariance
+ *
+ * The result of a chaos expansion
+ */
+
+class OT_API GaussianProcessConditionalCovariance
+  : public PersistentObject
+{
+  CLASSNAME
+
+public:
+
+  typedef GaussianProcessFitterResult::LinearAlgebra LinearAlgebra;
+  typedef Collection<CovarianceMatrix> CovarianceMatrixCollection;
+
+  /** Default constructor */
+  GaussianProcessConditionalCovariance();
+
+  /** Parameter constructor without any cholesky factor*/
+  GaussianProcessConditionalCovariance(const GaussianProcessRegressionResult & result);
+
+  /** Virtual constructor */
+  GaussianProcessConditionalCovariance * clone() const override;
+
+  /** String converter */
+  String __repr__() const override;
+  String __str__(const String & offset = "") const override;
+
+  /** Compute mean of new points conditionally to observations */
+  virtual Sample getConditionalMean(const Sample & xi) const;
+
+  /** Compute mean of new points conditionally to observations */
+  virtual Point getConditionalMean(const Point & xi) const;
+
+  /** Compute covariance matrix conditionally to observations*/
+  virtual CovarianceMatrix getConditionalCovariance(const Sample & xi) const ;
+
+  /** Compute covariance matrix conditionally to observations*/
+  virtual CovarianceMatrix getConditionalCovariance(const Point & xi) const;
+
+  /** Compute covariance matrices conditionally to observations (1 cov / point)*/
+  virtual CovarianceMatrixCollection getConditionalMarginalCovariance(const Sample & xi) const;
+
+  /** Compute covariance matrix conditionally to observations (1 cov of size outdimension)*/
+  virtual CovarianceMatrix getConditionalMarginalCovariance(const Point & xi) const;
+
+  /** Compute marginal variance conditionally to observations (1 cov of size outdimension)*/
+  virtual Scalar getConditionalMarginalVariance(const Point & point,
+      const UnsignedInteger marginalIndex = 0) const;
+
+  /** Compute marginal variance conditionally to observations (1 cov / point)*/
+  virtual Sample getConditionalMarginalVariance(const Sample & xi,
+      const UnsignedInteger marginalIndex = 0) const;
+
+  virtual Point getConditionalMarginalVariance(const Point & point,
+      const Indices & indices) const;
+
+  virtual Sample getConditionalMarginalVariance(const Sample & xi,
+      const Indices & indices) const;
+
+  /** Compute joint normal distribution conditionally to observations*/
+  virtual Normal operator()(const Sample & xi) const;
+
+  /** Compute joint normal distribution conditionally to observations*/
+  virtual Normal operator()(const Point & xi) const;
+
+  /** Method save() stores the object through the StorageManager */
+  void save(Advocate & adv) const override;
+
+  /** Method load() reloads the object from the StorageManager */
+  void load(Advocate & adv) override;
+
+
+protected:
+
+  void computePhi();
+  Matrix solveTriangularSystem(const Matrix & rhs) const;
```

Comment:
This method is currently uncovered by the current unit tests in the HMAT case (discovered by reading the output from CodeCov).

![image](https://github.com/openturns/openturns/assets/31351465/d01166fa-4631-4fa7-963c-f00e87c81881)

Reply:
As you can see the method is protected....